### PR TITLE
Capture StartEndRange information in SynExpr.AnonRecd to fix #2566

### LINKED ISSUE #2566: Idempotency problem when having a comment in an anonymous record and Stroustrup formatting
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAPMAvGMAD4A6ATmQHYRgCahYALAEyU10D0XYWiAZywdqJAL6QANOCgDsWAJbUA5gNRgA2jIig6dSABIsATwAOyNJCVYp2-QYBuAQ3QBXC5rt6ADNL17IAElqJGosAGUFAC9kP38IZi8wAF07MTidJMMTc3UrMNt4w2c3Dy14iABGDP9IAFknXAAZJUQmxBUsAAtCisrWbyTUvXS7XVqjMw9IDvgAeQAzFupEcJNMXoDHF3d1corWGoCAURDF5dikiEh0BcghtJrxrZzpgCNYWHRN%2BxLdtH28QAzEd9ME5BQsAAxJQuAByiAA7uhWj89FgKO4HiMnllJrlLB8vmioNtSnsrixQddwqYnNBEAAhRALWAURAABScFCcAFtsIgKCSIBisfFhnRRnpnvZXnkid9QcUdmVKQBWalQWn0pkstltWCIwXQJxyYIOWAmxSwajCviYxDYyW4or496fRV4v6qipgABsmsg2oZzNZ7IAqqZzBQTWbqBarQobXaFi45E6IFK6DLrm75R6ScryQDKQB2QPB3VhxAAYXQpoENZtQkx0CwbJTacd4seYzxcsJBaVZP%2Bnl9AA4K3SQ3r2XVELy3oLO%2Bh0z2cX3XQOoArCyOfRUAJxTnWh-VN9DJzWp1fd-wSzMuibbyC74feim%2Byq%2BSlB6eIABBBZ%2BAoJteV5Jw7VFO89AfMAs0yLcpnzYl3xVT8%2BmqX9KzPdlwgXBQ4EvW1ry7DN4KfF5kMHVCvXQksv0ObD-yAkD8N5QivivSloPIhDiH7aidyHOjizHPoQWYnUAIoWBXBCAARRAUQ4kCoIdPjKNlIT8hsNCxMBfxKmYQMGlwQIFgAFS6Dpwi6NksAAdQUeBujtQZ12dTdnx06w9w-Bi%2Bg1X8zIs6yOmOW87Ic5zXJ6TU-Q8%2B9e2lQSCSgPz9NHQy9EqAMQsaYIFgUXA5mjJx2woY5cFMdkBAEJMSMpccktglLszS6ZMtE7LKUqcsCtwAAlRA4AoeBYrczVmFaug4P4nNSRfbrXQC8T4kqSdBpGsb4DhVxF0FRZAn4Xk1E1SpNO8qj0sgXlXHQRQURWKE2QgrAQMs5CsoPDbj1-Ha2XgOoHqe1pXood61MDaAum5ekQIAfURFy3Ku1KkNulaJjWnK6AGUzGhknljDmCgWiESb4ua2aIHmrTc2Wgofow%2BJWCw316iJigSbJimsH2w6KGO07zr69GOsx6Z7sehRnsQCGocFL7chZwK2aYzniacUnyYUIQQdl%2BXFYq6Hf1h%2BG20FZHUZ6CXEJ8rHmZ637-FYSTObMgA1FVGSUeAlGUKm7Ra%2B2BKlvJsa2XHKVYEzBqheS20av2QkD4PptplJ2odm6uud1b6PWt3gs9xoFNgLAAHFsGq2rBAam0M5psPFuyHS3xd1m3fyz3Qbl1pGUvaAAGtGR5UfsAEOZqHCPlay%2BA6mt9XjPMfa7tNuzvC4M2OBs5hFkUH7AjQ6FXECU4rqAUa1qAAkJ5yFsWV40teKI3xmO5Enfet9Vgtq1iiZQ1BE7UGTjaSIwCKquHZJZWAwRQhYAqo1Fca5kobgxo7d0tEf6uz0Kwf6gCQLUGQQ4A0KhH5LgoJfJQN9GrPwqDeNBbUMGSywShT0uDu56CBD%2BPussLhND5G8eATg6ywAaioQ%2B8tUEwTmjncO7CaKcJxkXPGEAgQcwqJAOugoFD8jCC4AA0ogRApgEEdCwMEceTgwHU19EwuRdMFFtzzMo-yajKRAk1tooetiR4XAEDJOSIQERCEQMDfu8s651UbtQBh8RV7oK8pgvOHCPG719ECD2vjuS4QvvraAFB9Gwn4PAcM18bRKWgPWHkt9ZGt06uktWxceHx05ro4pBikHoDWLJVwLZXCmDWMYDYmokksJSWwtJlgo6-E8Vk0u2iTFmLMoLKhiw-HUACa0BJRlvBZ3ph-JaX8cGqMyRUIEvdtF9MIlgOoCBLgOLIm-MQMhkhxCsAIKEDV1COJAGIIAA)

### Original code

```fsharp
let x = {|
    Y = 42
    // test
|}

```
            
### Formatted code

```fsharp
let x = {|
    Y = 42
    // test

|}

```
            

### Reformatted code

```fsharp
let x = {|
    Y = 42
    // test


|}

```
            
### Problem description

Fantomas was not able to produce the same code after reformatting the result:

* Having a comment on the last line of an anonymous record leads to an extra empty line being added each time you format
* This is a relatively common scenario, i.e. using a comment temporarily to disable a certain field

### Workaround

Just don't do this in code.

### Extra information

- [ ] ~The formatted result breaks my code.~
- [ ] ~The formatted result gives compiler warnings.~
- [x] I or my company would be willing to help fix this.
- [x] Breaks Husky integration with precommit hook, the last line "walks down" on each format

### Options

Fantomas main branch at 2022-10-08T06:02:14Z - b0eea044f22051d63093de52b73fdd0d7f1a32eb

```fsharp
    { config with
                MultilineBlockBracketsOnSameColumn = true
                ExperimentalStroustrupStyle = true }
```

### Related example

A minor bug happens when `Stroustrup` is off. Note the comment being outdented and the closing `|}` bracket one space too far. Not sure it's related, see [example without Stroustrup](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAPMAvGMAD4A6ATmQHYRgCahYALAEyU10BaTZkAzlgoBLagHM%2BHWhABisWL0gAhAIZVIUugHotYLIkFSSAX0gAacFH7Ysosf1RgA2pYig6dSABIsATwAOyGiQoljmrp5eAG4q6ACuQc4RHgAMFh4ekACS1EjUWADKwgBeyOkZEMzJYAC6EcblbtXefoGOIfnhFd4x8YkuFRAAjI0ZkACyKrgAMqKI04jiWAAWXYNDrCnVdR4NEe5jPgGJkIvwAPIAZrPUiAV%2BmGuZ0bEJjgODrKOZAKK5VzcytUIJB0JdINt6qMDs9WicAEbydBPSK9N5oD4VADM308OWsFCwMlEsQAcogAO7oOYojxCBKQ3bQ5pHNrBRGwZG4nqvfrAli4kEFfwqaCIJSIS6wCiIAAKahUAFtsIgKLS6PTEIy6HsPDDInD2hyuSy0XzBmAAKyCqDC0XiyXS%2BawCmq6Aqaw5KKwd22WDUdUQTXaiC6uj6kGshFI9U8vrvfkANhtkDtYolUplAFV-IEKO7PdRvb7hP7A2BLrFrCGwGGmt0o0aY9yXvGMfyAOwptMOzOIADC6A9-H7-sEFDi0Cw0vLlfQ1YqOx1zIbhvZzdNvITFoAHN2RenHTLxohFfDVbOq1rF1D9iy11BjbHW%2BikhaAJz7%2B0Zp2j9Blm05wXDIl1DFdDgfSAnxbM1t3WNJ%2BVTA9EAAQUufQKFHRVFRUctgxvJk71XY4m05Z9YPbC0hhGRCex-GUClPYQ4H-ANAKvGs62Ie8SPXMiYK3Sj1i%2BWjkLQjDGMVZjOQA-l8JA289R4tlHw3BsKLfdYcVE%2B0UIoWA4lyAARRBqSkjC8Ina8FMIpTiJUjowgEttNIqIZmBTSZcCyS4ABVlkWAplmlLAAHVhHgFZyy2AjlyIiDeKgUJyME1yMiGa1EK8nz-MWH55zuYLCXCyLVhtRMYpsuK7IShzkuc19MXS5MsqmHJLmEXBzjzFRpwoH5cH8GV%2BH4Us2P5HdKo8UDa3A2FEsclKXKajwhi7VrcAAJUQOAKHgEqoptZgproGauIjKBG2CerN2W-khj3Dbtt2%2BBSTiM9VSuLJ9EVBwbSGTi5oNBbFTidBbGpW45AoHCsAw3ySIa811k-RDnuleBxjBiG5mh2GLJTaBlgVKdVQAfQpCKosB%2BL5rqzokbgipNk8qY9IoFRfHOChZkEA6yomk6IDOoHI0gm71NSla6FYGiLQmNmKA5rmeeEQQ3o%2Bigvp%2Bv77ppmq6ZOUHweESHEDx3r4cR27Gv5VgRPl9nOe53msCxk2zYtuGL0J4mOdJigKap1Z9fDZSTglw4NOliBWG0%2BWvIANV5JRRHgOx%2BfLSbQ-rWqI4Zm3keZjyNpkQypzG1Pcgz4Py2OnPuPs-OnMLpmMlYTKE6mIzYCwABxbABqGgxRv9TObWz2KwNp4GHOg1uhOZlqE%2Bx025iUf9oAAayUf2t%2BwfhzmoAolQHTl3vGi0gOs6bFLDpvSJNSW7otVh1vl8kqXX7BXUWBHAhMh1agwg-TUBQrkE8mtdYWnkrfWy9886PyWrbV%2Bj1HbUjENQMu1AK7%2BiKJg3qcQZS%2BVgDkPIWBepjUvAVBuF0WgLXns-FBnxUboIwtQShURnTiEgeeCggDRAgLGtAwY19aHhyQYzReGQsQIRXibQE0wlTwngCoQcsBRriE-mbahwE4HVQQYbSRC80oeCxHLQYkAh6qmEMqfIsQADSiBED%2BDIYsLAORd4qBwQLK%2BHEp6zRnmLBhako5S35FiB2liN7eK3oCfgekDK5HJIIRAmNV5myHsNUe1AREVFgadO%2BucjF8SfmEl%2BgwsTx2iWoeiiAjJq2gCIKSHD9DwCzMA-0JloBDg5qA3RN9CnwOKbPaM-ETExzAFiEu8trHNPcbEe4%2Bk4jjjiP4e4vhHjsRoQE86EjroFyYUXGRndLFOJcV5DWfCrgxOoHEuYeT0opCFrUIpjdEGlOQccsxy9LFLOYm7BAQI-E7KqqGSwNRyghH4DIUajgxHGCAA)

### Without Stroustrup example, before formatting

```
let x = {|
   Y = 42
   Z = "string"
   Foo = "Bar"
   // test
|}
```

### Without Stroustrup after formatting

```
let x =
    {| Y = 42
       Z = "string"
       Foo = "Bar"
    // test
     |}
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://fsprojects.github.io/fantomas/docs/end-users/IgnoreFiles.html)?</sub> *YES* 😃

### PR DESCRIPTION
**Pull request by dawedawe**
fixes the lost comments in #2566

### CODE REVIEW

File: src/Fantomas.Core/CodePrinter.fs [hunk 1]
```diff
@@ -2414,12 +2414,16 @@ and genMultilineAnonRecordAlignBrackets (isStruct: bool) fields copyInfo =
 
     let genAnonRecord =
         match copyInfo with
-        | Some ci -> sepOpenAnonRecd +> copyExpr fieldsExpr ci +> sepNln +> sepCloseAnonRecdFixed
+        | Some ci ->
+            sepOpenAnonRecd
```
Comment (nojaf) [inline src/Fantomas.Core/CodePrinter.fs:2418]: Would you mind doing the `genTriviaFor` thing as well for the openingBrace?
This isn't really required to solve the issues at hand but seems like a good thing to tag along

File: src/Fantomas.Core/CodePrinter.fs [hunk 2]
```diff
@@ -915,17 +915,17 @@ and genExpr synExpr ctx =
...
                 +> optSingle (fun e -> genExpr e +> !- " with ") copyInfo
                 +> col sepSemi fields genAnonRecordFieldName
-                +> sepCloseAnonRecd
+                +> genTriviaFor SynExpr_AnonRecd_ClosingBrace closingBrace (sepSpace +> sepCloseAnonRecdFixed)
 
             let longExpression =
                 ifAlignBrackets
-                    (genMultilineAnonRecordAlignBrackets isStruct fields copyInfo)
+                    (genMultilineAnonRecordAlignBrackets openingBrace isStruct fields copyInfo closingBrace)
                     (genMultilineAnonRecord isStruct fields copyInfo)
```
Comment (nojaf) [inline src/Fantomas.Core/CodePrinter.fs:929]: Please pass the information to `genMultilineAnonRecord` well.

File: src/Fantomas.Core/CodePrinter.fs [hunk 3]
```diff
@@ -915,17 +915,17 @@ and genExpr synExpr ctx =
                 let size = getRecordSize ctx xs
                 isSmallExpression size smallRecordExpr multilineRecordExpr ctx
 
-        | AnonRecord (isStruct, fields, copyInfo) ->
+        | AnonRecord (openingBrace, isStruct, fields, copyInfo, closingBrace) ->
             let smallExpression =
                 onlyIf isStruct !- "struct "
                 +> sepOpenAnonRecd
```
Comment (nojaf) [inline src/Fantomas.Core/CodePrinter.fs:921]: Use `genTrivia` in the `smallExpression` as well please.

File: src/Fantomas.Core/SourceParser.fs [hunk 4]
```diff
@@ -1071,7 +1071,8 @@ let (|Record|_|) =
 
 let (|AnonRecord|_|) =
     function
-    | SynExpr.AnonRecd (isStruct, copyInfo, fields, _) -> Some(isStruct, fields, Option.map fst copyInfo)
+    | SynExpr.AnonRecd (isStruct, copyInfo, fields, StartEndRange 1 (openingBrace, _, closingBrace)) ->
```
Comment (nojaf) [inline src/Fantomas.Core/SourceParser.fs:1074]: I think it should be `StartEndRange 2` here, as the size you pass to the active pattern is the character length of the token. For `{|` and `|}` that would be 2.

**Review by nojaf — CHANGES_REQUESTED** [review]
A good start here! Please dot some more i's and cross some t's and we can merge this in.

**Comment by dawedawe** [comment]
Adding `genTriviaFor` for the opening braces shows a separator problem:
In copy expressions:
```
let a =
    {| foo with// test1
```

In record instances:
```
let x =
    {| Y =// test1
```
The later could be solved be using
```genTriviaFor SynExpr_AnonRecd_OpeningBrace openingBrace sepOpenAnonRecdFixed```
instead of
```genTriviaFor SynExpr_AnonRecd_OpeningBrace openingBrace sepOpenAnonRecd```
but I'm really not sure if this is the way to go here.

**Comment by nojaf** [comment]
You could force the comment to be printed after `{|`, similar to how it is done for [normal records](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAPMAvGMGAPSkA6AThQHYQQCahYAjNXQL6QA04UAZ2xYAlrQDmA1GADa-CKAYNIAEiwBPAA7I0kMVl7zlKgG4BDdAFcdso0oAMfJUsgBJWklpYAyiIBeyE7OEAAsdmAAukacQQrhqhra0npehsGq5lY2csEQrLHOkACyZrgAMmKIZYgSWAAWabmsAEz24VFKMUaKhWpaNpA18ADyAGYVtIjeGpiNLqYW1tI5uc0FLgCiHmMTgeEQkOijkO3RBT3ziQMARrCw6HPGmUtoK8EAzOvK7kJUWABiYgsADlEAB3dCVR5KLBUaynTrneJ9JK6W73aFQBZZZb7MAhL4HbyaMzQRAAIUQo1gVEQAAUzFQzABbbCIKiYiCw%2BHBDoMLpKC7GK7JdEPL4ZRbZPEAVkJUGJpIpVJpVVgYPZ0DMQncJlgWtEsFonLA3MQCP5SPSKJud3FyOe0tyYAAbPLIIqyZTqbSAKqabRULU62h6g0iI0m0YWIQWiAChhCg420V2zGSnGvPEAdndnuVPsQAGF0NqBEWjQJudAsDSozHzbyzt1kSK0WmJdiXrZnQAOPMkr0q2lFRDM67s%2BvoWNNxEt61tqBi9Ndp25ACcA6V3tVFfQkfl0enjecfPjVt6i8gy87jtxztYjjxHsHiAAgqMsOyK8zmWYTWacZgAmcQLv0qYYreUr3k0%2BTPvmO60t4Y4iHA%2B7GoeDZASBxCtuB7aQQ60FZg%2Bazwa%2BH5flQyHMqh9wHnigGzpa86XvhS4dkRmY9k0nzkUqb5ULAlgeAAIogkK0VRAFwieShnsBF6XOxKQGFB3FvM4rAEs%2BJS4K4owACp1DU3h1DSWAAOoiPA9Qmm0zHnqxymolA%2BgrneJFNHKumlAZxk1Bsx5mRZ1m2Q08oug5p7NoKeGuapHnETxwSsG6vn6bQowiLgwyBmYtZUBsuCaLSAgCBGGF4r20XybFibxQM7nqd2mlKKwuYZQASogcBUPAYV2fKIS1QwCk4UmWJXs1XGtXirD9t1vU0vAwKWOO7JjK4X7MlI8qsNhSnCipzKWOgoiQpM-w0n%2BWBUYZ4EtWuqWbs%2BPV9fARRnRdlTXVQt3Se60B1IypJUQA%2BmCNl2YdznHQlM3Wp5KXOK07p6YJTLqMMVAVFWg0RdVo0QONR3JtNqRPTBwTNHBzrFKUmNmNjuMiFWa0bVQW07Xt82w3FYEJad50iJdiB-QD7IPUkVNeTTZH00zLN41gX0i2LEsFYDz7A6DNbspD0MNPzDWC01lOzc9qN8fTekAGpSuSYjwGI4gEyaNUm6BbEIxbSPJW1DDNDptulP8Ik1pVTseK77vDcTkT1d7Lnm2plvU6jPmh7gomwFgADi2DFaViDlZVcdE17uFmxB9r%2BxpeLNOltvfaLlTkvu0AANbkky3fYAIwy0N4LLFvc61Vc6R4zjFc4Cz7tqEfXc3Os0nX06CELt9gGo1NLiDidltAiIatBvh4o6c7zzpMbPLHzyntdJQ3q%2BLYrkLiLQ4e0JHRq%2BJ-BVLC0kMrAdwngsAFUqlOGedU56mwXk-WWKMlDNFeu-KitBIEmDVBIS%2BE4qCHzECfSq19cjTzkmNJO1cEEETrr0ZGgcIDvCfC3EWuwygsmuPAMwJZYAVQkJvMW0CKEkyoZNBIKkbzpzls4d4dNciQGLuyEQrIvAWAANKIEQJoMBNQsDuF7mYH%2BhMp5YUcopOG5NJGcWXlbJQ7wFYKI7kYruuwBCCWEh4UEVZECfVbmLYuZUKqVhkjyO%2BTkH7w0XnQ%2BYDC8TvBtk4xkiED5s2gFQFRQIvzwF9MfI04loCliZKfYRVdxEploc-FeuR3ghwUUojJqiIHoGmEJSwVY4SaGmOoWYmFjxlMaskRG9CA7xKzgozR2i9Ic3wWMZxtBXGVFIalewCdSaWKmtYpeIyX41Obgo1pqFVYID2KY-p5jOD8AiEEPQAh-gVWkOQkAnAgA).

We can merge this PR as is, for my part. It's up to you.

**Comment by dawedawe** [comment]
> You could force the comment to be printed after `{|`, similar to how it is done for [normal records](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAbRAXMAPMAvGMGAPSkA6AThQHYQQCahYAjNXQL6QA04UAZ2xYAlrQDmA1GADa-CKAYNIAEiwBPAA7I0kMVl7zlKgG4BDdAFcdso0oAMfJUsgBJWklpYAyiIBeyE7OEAAsdmAAukacQQrhqhra0npehsGq5lY2csEQrLHOkACyZrgAMmKIZYgSWAAWabmsAEz24VFKMUaKhWpaNpA18ADyAGYVtIjeGpiNLqYW1tI5uc0FLgCiHmMTgeEQkOijkO3RBT3ziQMARrCw6HPGmUtoK8EAzOvK7kJUWABiYgsADlEAB3dCVR5KLBUaynTrneJ9JK6W73aFQBZZZb7MAhL4HbyaMzQRAAIUQo1gVEQAAUzFQzABbbCIKiYiCw%2BHBDoMLpKC7GK7JdEPL4ZRbZPEAVkJUGJpIpVJpVVgYPZ0DMQncJlgWtEsFonLA3MQCP5SPSKJud3FyOe0tyYAAbPLIIqyZTqbSAKqabRULU62h6g0iI0m0YWIQWiAChhCg420V2zGSnGvPEAdndnuVPsQAGF0NqBEWjQJudAsDSozHzbyzt1kSK0WmJdiXrZnQAOPMkr0q2lFRDM67s%2BvoWNNxEt61tqBi9Ndp25ACcA6V3tVFfQkfl0enjecfPjVt6i8gy87jtxztYjjxHsHiAAgqMsOyK8zmWYTWacZgAmcQLv0qYYreUr3k0%2BTPvmO60t4Y4iHA%2B7GoeDZASBxCtuB7aQQ60FZg%2Bazwa%2BH5flQyHMqh9wHnigGzpa86XvhS4dkRmY9k0nzkUqb5ULAlgeAAIogkK0VRAFwieShnsBF6XOxKQGFB3FvM4rAEs%2BJS4K4owACp1DU3h1DSWAAOoiPA9Qmm0zHnqxymolA%2BgrneJFNHKumlAZxk1Bsx5mRZ1m2Q08oug5p7NoKeGuapHnETxwSsG6vn6bQowiLgwyBmYtZUBsuCaLSAgCBGGF4r20XybFibxQM7nqd2mlKKwuYZQASogcBUPAYV2fKIS1QwCk4UmWJXs1XGtXirD9t1vU0vAwKWOO7JjK4X7MlI8qsNhSnCipzKWOgoiQpM-w0n%2BWBUYZ4EtWuqWbs%2BPV9fARRnRdlTXVQt3Se60B1IypJUQA%2BmCNl2YdznHQlM3Wp5KXOK07p6YJTLqMMVAVFWg0RdVo0QONR3JtNqRPTBwTNHBzrFKUmNmNjuMiFWa0bVQW07Xt82w3FYEJad50iJdiB-QD7IPUkVNeTTZH00zLN41gX0i2LEsFYDz7A6DNbspD0MNPzDWC01lOzc9qN8fTekAGpSuSYjwGI4gEyaNUm6BbEIxbSPJW1DDNDptulP8Ik1pVTseK77vDcTkT1d7Lnm2plvU6jPmh7gomwFgADi2DFaViDlZVcdE17uFmxB9r%2BxpeLNOltvfaLlTkvu0AANbkky3fYAIwy0N4LLFvc61Vc6R4zjFc4Cz7tqEfXc3Os0nX06CELt9gGo1NLiDidltAiIatBvh4o6c7zzpMbPLHzyntdJQ3q%2BLYrkLiLQ4e0JHRq%2BJ-BVLC0kMrAdwngsAFUqlOGedU56mwXk-WWKMlDNFeu-KitBIEmDVBIS%2BE4qCHzECfSq19cjTzkmNJO1cEEETrr0ZGgcIDvCfC3EWuwygsmuPAMwJZYAVQkJvMW0CKEkyoZNBIKkbzpzls4d4dNciQGLuyEQrIvAWAANKIEQJoMBNQsDuF7mYH%2BhMp5YUcopOG5NJGcWXlbJQ7wFYKI7kYruuwBCCWEh4UEVZECfVbmLYuZUKqVhkjyO%2BTkH7w0XnQ%2BYDC8TvBtk4xkiED5s2gFQFRQIvzwF9MfI04loCliZKfYRVdxEploc-FeuR3ghwUUojJqiIHoGmEJSwVY4SaGmOoWYmFjxlMaskRG9CA7xKzgozR2i9Ic3wWMZxtBXGVFIalewCdSaWKmtYpeIyX41Obgo1pqFVYID2KY-p5jOD8AiEEPQAh-gVWkOQkAnAgA).

Ah thanks, that was way easier than I thought.

File: src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs [hunk 5]
```diff
@@ -1395,3 +1395,61 @@ let f
...
+    {|
+        Y = 42
+    // test
+    |}
+
+let z = {| Y = 42 |}
+
+let a =
+    {| // test1
+    foo with
```
Comment (nojaf) [inline src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs:1450]: I know it is the same as we have it for regular records.
But in the future, we might want to have something like
```fsharp
{| // foo
   bar with
    A = 1
    B = 2 |}
```

I don't feel strongly about this, so no need to address this in this PR.

Comment (dawedawe) [reply src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs:1450]: I think this should better be handled in a separate PR.

Comment (nojaf) [reply src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs:1450]: Sounds good to me!
I have no idea why the CI is failing, the NuGet lock file should not be throwing errors.
I think we need to apply the steps explained [here](https://github.com/NuGet/Home/issues/7921#issuecomment-478152479).
I'll do that in a separate PR so you can rebase with main.

Comment (nojaf) [reply src/Fantomas.Core.Tests/MultilineBlockBracketsOnSameColumnRecordTests.fs:1450]: There is also https://github.com/dotnet/fsharp/issues/13678.
I hope it is ok.

**Pull request merged**